holo-ospf/src/packet/tlv.rs

```diff
@@ -534,6 +556,42 @@ impl From<RouterFuncCaps> for RouterFuncCapsTlv {
     }
 }
 
+// ===== impl DynamicHostnameTlv ====
+
+impl DynamicHostnameTlv {
+    pub(crate) fn decode(tlv_len: u16, buf: &mut Bytes) -> DecodeResult<Self> {
+        let mut hostname = String::new();
+        for _ in 0..tlv_len {
+            let c = buf.get_u8();
+            if c == 0 {
+                break;
+            }
+            hostname.push(c as char);
+        }
+
+        Ok(DynamicHostnameTlv { hostname })
+    }
+
+    pub(crate) fn encode(&self, buf: &mut BytesMut) {
+        let start_pos =
+            tlv_encode_start(buf, RouterInfoTlvType::DynamicHostname);
+        let mut hostname = self.hostname.clone();
```

Comment:
you could try something like this to avoid cloning the string:
```rs
        let hostname = self.hostname.as_bytes();
        let hostname_len = usize::min(hostname.len(), MAX_HOSTNAME_LEN);
        buf.put_slice(&hostname[..hostname_len]);
```